src/topp/FLASHDeconv.cpp

```diff
@@ -388,20 +411,34 @@ class TOPPFLASHDeconv : public TOPPBase
     }
     // Max MS Level is adjusted according to the input dataset
     current_max_ms_level = current_max_ms_level > max_ms_level ? max_ms_level : current_max_ms_level;
-
     // Run FLASHDeconv here
 
     int num_last_deconvolved_spectra = getIntOption_("preceding_MS1_count");
+    if (!in_log_file.empty())
+    {
+      num_last_deconvolved_spectra = 50; // if FLASHIda log file exists, keep up to 50 survey scans.
+    }
 
     auto last_deconvolved_spectra = std::unordered_map<UInt, std::vector<DeconvolvedSpectrum>>();
-    MSExperiment exp, exp_annotated;
+    MSExperiment exp(map), exp_annotated(map);
+    exp.clear(false);
+    exp_annotated.clear(false);
 
     auto fd = FLASHDeconvAlgorithm();
-    FLASHDeconvAlgorithm fd_charge_decoy, fd_noise_decoy, fd_iso_decoy;
+    FLASHDeconvAlgorithm fd_charge_dummy, fd_noise_dummy, fd_iso_dummy;
 
     Param fd_param = getParam_().copy("Algorithm:", true);
     DoubleList tols = fd_param.getValue("tol");
 
+    if(target_precursor_mz > 0)
```

Comment:
please introduce a bool variable is_targeted_study.
easier to read and reason
e.g., the description of the tools says that target_precursor_mz is ignored if target_precursor_charge is set to 0
this leads to a bit convoluted logic because here it sets parameters even in the case that the charge is set to 0. As a reader I don't know if it get's ignored later down the line etc.

Better to make these things more explicit: have a bool that is true if FD is run in targeted mode and then use that to decide how to configure the underlying FDAlgo

Reply:
I see. that is a good tip. I will fix this in the next revision. Now target_precursor_mz and target_precursor_charge act together. Some users specify only charge and others both. But FLASHDeconv.cpp need to be moved to ANANYSIS/TOPDOWN and refactored.